Marlin/Marlin_main.cpp

```diff
@@ -4279,6 +4282,144 @@ inline void gcode_G92() {
+//  goes through a reset which sets all I/O pins to tristate)
+ 
+// PWM duty cycle goes from 0 (off) to 255 (always on) 
+ 
+ inline void gcode_M3() {
+
+// this is the "pretty" version.  It accepts RPM as input
+ 
+  #ifdef SPINDLE_SPEED_ENABLED
+  
+  #define RPM_slope  118.4319437   // use gcode_M4() to get the points needed to compute the RPM vs OCRxx_VAL line
+  #define RPM_intercept 0
+  
+     float spindle_speed = code_seen('S') ? code_value_float() : 0; 
+     if (spindle_speed == 0) { 
+         digitalWrite(SPINDLE_ON_OFF_PIN, !SPINDLE_ON_OFF_INVERT);  //turn spindle off (active low);
+     }
+      else {
+         int8_t OCRxx_VAL = spindle_speed/RPM_slope + RPM_intercept;      // convert RPM to PWM duty cycle
+         if (spindle_speed <= 5000) OCRxx_VAL = 42; // minimum the SuperPid will go
+         if (spindle_speed >= 30000) OCRxx_VAL = 255; // maximum the SuperPid will go
+         #if SPINDLE_SPEED_INVERT==true
+            OCRxx_VAL = 255 - OCRxx_VAL ;
+         #endif         
+         digitalWrite(SPINDLE_ON_OFF_PIN, SPINDLE_ON_OFF_INVERT);  // turn spindle on (active low)
+         pinMode(SPINDLE_SPEED_PIN, OUTPUT);
+         analogWrite(SPINDLE_SPEED_PIN,OCRxx_VAL); 
+         } 
+   #else
+    digitalWrite(SPINDLE_ON_OFF_PIN, SPINDLE_ON_OFF_INVERT);  //turn spindle on (active low) if spindle speed option not enabled
+  #endif
+ } 
+ 
+inline void gcode_M4() {
+
+/* this one is used for debug
+   it accepts inputs of 0-255
+   
+   To use it you'll need to change the M4 command from "gcode_M3()" to "gcode_M4()"
+   
+   This was used to get the RPM vs. OCRxx PWM values
+     That was thrown into Excel to determine a best fit line
+     for use in the M3 routine.
+
+  
+ */
+ /*
+SERIAL_PROTOCOLLNPAIR("TCCR2A:  ", TCCR2A);
+SERIAL_PROTOCOLLNPAIR("TCCR2B:  ", TCCR2B); 
+SERIAL_PROTOCOLLNPAIR("TCCR3A:  ", TCCR3A);
+SERIAL_PROTOCOLLNPAIR("TCCR3B:  ", TCCR3B); 
+SERIAL_PROTOCOLLNPAIR("TCCR4A:  ", TCCR4A);
+SERIAL_PROTOCOLLNPAIR("TCCR4B:  ", TCCR4B); 
+SERIAL_PROTOCOLLNPAIR("TCCR5A:  ", TCCR5A);
+SERIAL_PROTOCOLLNPAIR("TCCR5B:  ", TCCR5B);  
+ */
+ 
+  #ifdef SPINDLE_SPEED_ENABLED
+     uint8_t spindle_speed = code_seen('S') ? code_value_byte() : 0; 
+     if (spindle_speed == 0) { 
+         digitalWrite(SPINDLE_ON_OFF_PIN, !SPINDLE_ON_OFF_INVERT);  //turn spindle off (active low);
+     }
+      else {
+         digitalWrite(SPINDLE_ON_OFF_PIN, SPINDLE_ON_OFF_INVERT); //turn spindle on (active low) if spindle speed option not enabled
+         pinMode(SPINDLE_SPEED_PIN, OUTPUT);
+         #if SPINDLE_SPEED_INVERT==true
+            spindle_speed = 255 - spindle_speed ;
+         #endif         
+         
+         analogWrite(SPINDLE_SPEED_PIN,spindle_speed); 
+//SERIAL_PROTOCOLLNPGM("SET spindle_speed: ");         
+         } 
+     
+   #else
+    digitalWrite(SPINDLE_ON_OFF_PIN, SPINDLE_ON_OFF_INVERT);  //turn spindle on (active low) if spindle speed option not enabled
```

Comment:
For this you can use fastio:
```cpp
WRITE(SPINDLE_ON_OFF_PIN, SPINDLE_ON_OFF_INVERT)
```
Consider `SPINDLE_ENABLE_PIN` as an alternative name for this pin, and `SPINDLE_ENABLE_INVERT` for the option. Generally '`ENABLE`' is used for this kind of pin.